### PR TITLE
fix bugs in api_regex, reading from and writing to json

### PR DESCRIPTION
**Pull request by CoderHam**
These fixes were made in v2test. Bringing them over to master

### CODE REVIEW

File: src/servers/http_server_v2.cc [hunk 1]
```diff
@@ -715,25 +715,39 @@ WriteDataToJsonHelper(
...
     const DataType dtype)
 {
-  for (int i = 0; i < shape[shape_index].GetInt(); i++) {
-    if ((shape_index + 1) != (int)shape.Size()) {
-      rapidjson::Value response_output_array(rapidjson::kArrayType);
-      WriteDataToJsonHelper(
-          &response_output_array, allocator, shape, shape_index + 1, base,
-          counter, dtype);
-      response_output_val->PushBack(response_output_array, allocator);
+  if (shape_index == (int)shape.Size()) {
```
Comment (deadeyegoodwin) [inline src/servers/http_server_v2.cc:718]: Does shape.Size() return a size_t? Why is 'shape_index' type "int"? If it is an index size_t would be more appropriate.

File: src/servers/http_server_v2.cc [hunk 2]
```diff
@@ -715,25 +715,39 @@ WriteDataToJsonHelper(
...
+    if (dtype != TYPE_STRING) {
+      rapidjson::Value data_val((T)(base[*counter]));
```
Comment (deadeyegoodwin) [inline src/servers/http_server_v2.cc:720]: base is already a T*, cast (T) is unnecessary.

File: src/servers/http_server_v2.cc (hunk 1 above)
Comment (CoderHam) [reply src/servers/http_server_v2.cc:718]: That is right. Will fix this

File: src/servers/http_server_v2.cc (hunk 2 above)
Comment (CoderHam) [reply src/servers/http_server_v2.cc:720]: Correct. Will fix in a follow up PR

**Pull request merged**